mesop/server/server.py

```diff
@@ -132,7 +133,17 @@ def generate_data(ui_request: pb.UiRequest) -> Generator[str, None, None]:
         else:
           yield from render_loop(path=ui_request.path, init_request=True)
       elif ui_request.HasField("user_event"):
-        runtime().context().update_state(ui_request.user_event.states)
+        event = ui_request.user_event
+        if event.HasField("navigation") or event.HasField("resize"):
+          if event.HasField("resize"):
+            runtime().context().set_viewport_size(event.resize.viewport_size)
+          elif event.HasField("navigation") and event.navigation.HasField(
```

Comment:
[question] What are the cases where viewport_size is not included with the navigation event?

Reply:
viewport_size isn't included in a regular navigation (me.navigate), but it's included in navigations triggered by a hot reload.

I've treated hot reload as a type of navigation b/c it uses (mostly) the same code paths, but it makes this a bit more confusing. I can refactor this later.